wsimod/nodes/nodes.py

```diff
@@ -88,6 +88,19 @@ def apply_overrides(self, overrides: Dict[str, Any] = {}) -> None:
         Args:
             overrides (dict, optional): Dictionary of overrides. Defaults to {}.
         """
+        # overrides data_input_dict
+        from wsimod.orchestration.model import read_csv
+
+        content = overrides.pop("data_input_dict", self.data_input_dict)
+        if isinstance(content, str):
+            self.data_input_dict = read_csv(content)
+        elif isinstance(content, dict):
+            self.data_input_dict = content
```

Comment:
Why these two cases? Wouldn't it be best to tell the user to put the data input dict directly in the overrides? Or only in an external file. 

I might be a bit cynic here, but I think that the more freedom you give a user with the configuration, the higher the chance for them to do it wrong.

Reply:
The reason behind is that I don't know what preferences the users might have. I personally think directly using read_csv would be a more reasonable way - users can specify the timeseries csv files and only provide the address here. Our colleagues often have problems in directly changing the data_input_dict because the timestamp in the key of the dict (e.g., `(precipitation, 2000-01-01)`) is a new class defined in WSIMOD, so that they often get the error: `key *** is not found` when they specify the timstamp as a normal datetime. I don't know whether @barneydobson has a preference here.